internal/gengapic/auxiliary.go

```diff
@@ -0,0 +1,365 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	longrunning "cloud.google.com/go/longrunning/autogen/longrunningpb"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// auxTypes gathers details of types we need to generate along with the client
+type auxTypes struct {
+	// Map of RPC descriptor to assigned operationWrapper.
+	methodToWrapper map[*descriptorpb.MethodDescriptorProto]operationWrapper
+
+	// RPC-specific operation wrapper types by wrapper name e.g. CreateFooOperation.
+	opWrappers map[string]operationWrapper
+
+	// "List" of iterator types. We use these to generate FooIterator returned by paging methods.
+	// Since multiple methods can page over the same type, we dedupe by the name of the iterator,
+	// which is in turn determined by the element type name.
+	iters map[string]*iterType
+
+	customOp *customOp
```

Comment:
nit: rest of the fields have docs, could you add a note here as well.

Reply:
Sure thing, done.